wom/src/test/scala/wom/types/WomFileTypeSpec.scala

```diff
@@ -0,0 +1,124 @@
+  )
+
+  forAll(coercionTests) { (description, womFileType, value, expected) =>
+    it should s"coerce $description" in {
+      womFileType.coerceRawValue(value).get should be(expected)
+      womFileType.coercionDefined(value) should be(true)
+    }
+  }
+
+  lazy val failedCoercionTests = Table(
+    ("description", "womFileType", "value", "expected"),
+
+    ("a double to a dir", WomSingleDirectoryType, 6.28318,
+      "No coercion defined from '6.28318' of type 'java.lang.Double' to 'Dir'."),
+    ("a double to a file", WomSingleFileType, 6.28318,
+      "No coercion defined from '6.28318' of type 'java.lang.Double' to 'File'."),
+    ("a double to a glob", WomGlobFileType, 6.28318,
+      "No coercion defined from '6.28318' of type 'java.lang.Double' to 'Glob'."),
+
+    ("a js number to a dir", WomSingleDirectoryType, JsNumber(6.28318),
+      "No coercion defined from '6.28318' of type 'spray.json.JsNumber' to 'Dir'."),
+    ("a js number to a file", WomSingleFileType, JsNumber(6.28318),
+      "No coercion defined from '6.28318' of type 'spray.json.JsNumber' to 'File'."),
+    ("a js number to a glob", WomGlobFileType, JsNumber(6.28318),
+      "No coercion defined from '6.28318' of type 'spray.json.JsNumber' to 'Glob'."),
+
+    ("a wom float to a dir", WomSingleDirectoryType, WomFloat(6.28318),
+      "No coercion defined from '6.28318' of type 'Float' to 'Dir'."),
+    ("a wom float to a file", WomSingleFileType, WomFloat(6.28318),
+      "No coercion defined from '6.28318' of type 'Float' to 'File'."),
+    ("a wom float to a glob", WomGlobFileType, WomFloat(6.28318),
+      "No coercion defined from '6.28318' of type 'Float' to 'Glob'."),
+
+    ("a wom dir to a file", WomSingleDirectoryType, WomSingleFile("example/file"),
+      """No coercion defined from '"example/file"' of type 'File' to 'Dir'."""),
+    ("a wom dir to a glob", WomSingleDirectoryType, WomGlobFile("example/glob*"),
+      """No coercion defined from 'glob("example/glob*")' of type 'Glob' to 'Dir'."""),
+
+    ("a wom file to a dir", WomSingleFileType, WomSingleDirectory("example/dir"),
+      """No coercion defined from '"example/dir"' of type 'Dir' to 'File'."""),
+    ("a wom file to a glob", WomSingleFileType, WomGlobFile("example/glob*"),
+      """No coercion defined from 'glob("example/glob*")' of type 'Glob' to 'File'."""),
+
+    ("a wom glob to a dir", WomGlobFileType, WomSingleDirectory("example/dir"),
+      """No coercion defined from '"example/dir"' of type 'Dir' to 'Glob'."""),
+    ("a wom glob to a file", WomGlobFileType, WomSingleFile("example/file"),
+      """No coercion defined from '"example/file"' of type 'File' to 'Glob'.""")
+  )
+
+  forAll(failedCoercionTests) { (description, womFileType, value, expected) =>
+    it should s"fail to coerce $description" in {
+      womFileType.coerceRawValue(value).failed.get.getMessage should be(expected)
```

Comment:
I believe there's a `should throwWithExceptionMessage()` or some such in scalatest